machine/v1beta1/types_azureprovider.go

```diff
@@ -83,6 +86,11 @@ type AzureMachineProviderSpec struct {
 	// SecurityProfile specifies the Security profile settings for a virtual machine.
 	// +optional
 	SecurityProfile *SecurityProfile `json:"securityProfile,omitempty"`
+	// UltraSSDCapability enables or disables Azure UltraSSD capability for a virtual machine.
+	// When omitted, the platform may enable the capability based on the configuration of data disks.
+	// +kubebuilder:validation:Enum:="Enabled";"Disabled"
+	// +optional
+	UltraSSDCapability AzureUltraSSDCapabilityState `json:"ultraSSDCapability,omitempty"`
```

Comment:
Is this related to the addition of datadisk?  The fact that the platform can set this almost suggests that certain datadisks should beinvalid unless this is enabled?

Reply:
If you want to use an UltraSSD as a PersistentVolume, then the capability needs to be enabled on the VM, as such we need a way for users to specify that they want it turned on when they don't have an UltraSSD directly attached.

We mention in the enhancement that if you set this to `Disabled` and specify an UltraSSD data disk, then that's invalid and we will reject the manifest. If you don't set a value and request an UltraSSD, we will enable it for you implicitly so that the disk is valid

Reply:
Update the comment to explain the PV case and to explain the failure mode if set to an impossible value.

Also, be sure to validate that someone isn't setting to disabled and trying to add an ultra disk.